addons/gut/gut.gd

```diff
@@ -760,6 +762,10 @@ func _test_the_scripts(indexes=[]):
 					_lgr.risky(str(_current_test.name, ' did not assert'))
 
 				_current_test.has_printed_name = false
+				
+				var us_elapsed = Time.get_ticks_usec() - ticks_before
+				_current_test.time_taken = us_elapsed / 1000000.0
```

Comment:
Wasn't sure what `us_elapsed` meant at first.  Just make this a single line.
```
_current_test.time_taken = (Time.get_ticks_usec() - us_elapsed) / 1000000.0
```

Then there's the dumb debate going on in my head as to whether the division should actually happen when making the xml and usecs should actually be stored in `time_taken`.  Doing the division here is fine, but it does feel a bit "wrong".  Having it as a single line somehow makes it less "wrong".  It's fine.  Right?  Yea, it's fine.  I've already wasted more time thinking about this incredibly minor detail and I'm moving on.

Reply:
I don't know if this helps with the quandary, but the schema here has the time output in seconds: https://github.com/windyroad/JUnit-Schema/blob/master/JUnit.xsd
Although I do understand the issue with that, and source of the quandary. 

But thank you so much for the detailed comments. I'll make this change and add the tests 😄

Reply:
Yea, I agree the time should go into the XML as seconds.  For some reason I want to do the conversion to seconds when the XML is being created, and storing microseconds in `time_taken`.  It's easier to test if we store seconds though, since we can test for that much easier in `test_result_exporter.gd` than testing the resulting XML.

It's just one of those minor details that my brain wants to overthink.  Storing seconds is the right answer.

Reply:
I've added the time taken tests now. I haven't looked at the XML verification yet. Finding a more accurate way of getting the time was tricky - the standard `await x.timeout` approaches are incredibly inaccurate at small values. I added this in the `test_time_taken.gd` resource test script to compensate for it. It still logs the wait time to the logger like `await wait_seconds`.

```gdscript
func more_accurate_wait_time(time_to_wait_msec : int)->void:
	gut._lgr.yield_msg(str('-- Awaiting ', time_to_wait_msec / 1000.0, ' second(s) -- '))
	var start := Time.get_ticks_msec()
	while (Time.get_ticks_msec() - start < time_to_wait_msec):
		await get_tree().process_frame

func test_pass_time_taken_about_half_s():
	await more_accurate_wait_time(500)
	assert_eq('one', 'one')
```

Reply:
I've added the time taken tests now. I haven't looked at the XML verification yet. Finding a more accurate way of getting the time was tricky - the standard `await x.timeout` approaches are incredibly inaccurate at small values. I added this in the `test_time_taken.gd` resource test script to compensate for it. It still logs the wait time to the logger like `await wait_seconds`.

```gdscript
func more_accurate_wait_time(time_to_wait_msec : int)->void:
	gut._lgr.yield_msg(str('-- Awaiting ', time_to_wait_msec / 1000.0, ' second(s) -- '))
	var start := Time.get_ticks_msec()
	while (Time.get_ticks_msec() - start < time_to_wait_msec):
		await get_tree().process_frame

func test_pass_time_taken_about_half_s():
	await more_accurate_wait_time(500)
	assert_eq('one', 'one')
```

Reply:
In cc27eec8684978f2ee9bbe27df073748c51d6aaa I modified the xml validator to also checks for valid tags and attributes. It doesn't check for valid nesting / structure - just that the tags that are encountered are expected, and that those tags contain only the required attributes for that tag.